pkg/services/ngalert/state/historian/loki.go

```diff
@@ -242,14 +243,17 @@ func statesToStream(rule history_model.RuleMeta, states []state.StateTransition,
 			continue
 		}
 
+		sanitizedLabels := removePrivateLabels(state.Labels)
+		instFingerprint, _ := json.Marshal(sanitizedLabels)
```

Comment:
I don't think this is correct.

There's no sorting of the labels so so the same labels might produce different results.

On top of that, why do we need the json marshalling? (which is _very_ expensive - probably more than the hashing itself).

I'd replace the current approach with:

```
md5.Sum([]byte(sanitizedLabels.String()))
```

Which a) would remove the need for JSON marshalling and then have a guaranteed order.

Reply:
The json package sorts the map keys automatically - this is actually why I originally chose json, because it's stable. `String()` felt more unreliable to me.

I instead decided to lift the exact same fingerprint calculation that Prometheus implements. It's a super simple FNV hash with no intermediate step, so it's faster than all other suggestions - and it also sorts keys, and has the side effect of producing the same fingerprint across alertmanagers :smile: . There was no need for a crypto hash here.

Reply:
lifting FNV is something I can def get behind - historically, JSON computations are pretty expensive in Go and I'd to shy away from it when possible. 

Out of curiosity though, why does `String()` feels unreliable to you? In my eyes, this looks pretty standard.